iFixit/react-commerce
# Playwright: Add Supporting Infrastructure

## Description

This updates our test infrastructure to support writing and running Playwright tests. We will address integrating Playwright into our CI workflow in later pulls, for now, this will add the basic setup to begin writing tests and running them locally. As we continue the migration process, #1111, we will modify the `Playwright` config file to fit our testing needs. 

## QA Notes:
You can add the following test to `Playwright` to assert the expected behavior:
```Typescript
import { test, expect } from '@playwright/test';

test('homepage has title and links to intro page', async ({ page }) => {
  await page.goto('https://playwright.dev/');

  // Expect a title "to contain" a substring.
  await expect(page).toHaveTitle(/Playwright/);

  // create a locator
  const getStarted = page.getByRole('link', { name: 'Get started' });

  // Expect an attribute "to be strictly equal" to the value.
  await expect(getStarted).toHaveAttribute('href', '/docs/intro');

  // Click the get started link.
  await getStarted.click();

  // Expects the URL to contain intro.
  await expect(page).toHaveURL(/.*intro/);
});

```

- [ ] Confirm `Cypress` and `Jest` do not pick up tests in the `playwright` subfolder
- [ ] Confirm `Cypress` and `Jest` tests still work
- [ ] Confirm running `pnpm playwright:run` will pick up and run tests in the `playwright` subfolder
	- [ ] Confirm that you do not need the server running for the `Playwright` tests to be ran
	- [ ] Confirm that if you do have a server running, the `Playwright` tests will still work

Closes: #1112
